### PR TITLE
A set of fixes in porting from C to C++

### PR DESCRIPTION
**Pull request by apanda**
I can break this into smaller pull requests if that helps, but the code changes should be quite minimal. The main changes of interest are:

- PMD ports were unusable with PCI specs, this change reallows that. I don't think the solution is particularly elegant, but I had limited options because of choices made by protobuf.

- The rewrite module reliably segfaulted for me, fixed that by changing memcpy used.

- Measure did not forward most packets, so a minor fix for that module.

- The histogram bits in utils were unusable.

### CODE REVIEW

File: core/modules/measure.cc [hunk 1]
```diff
@@ -38,50 +38,48 @@ void Measure::ProcessBatch(bess::PacketBatch *batch) {
...
+        bytes_cnt_ += batch->pkts()[i]->total_len();
+        total_latency_ += diff;
 
-      record_latency(&hist_, diff);
+        record_latency(&hist_, diff);
+      }
     }
+    RunNextModule(batch);
   }
-}
```
Comment (melvinw) [inline core/modules/measure.cc]: this should come back. travis probably already yelled about this

File: core/modules/measure.cc [hunk 2]
```diff
@@ -38,50 +38,48 @@ void Measure::ProcessBatch(bess::PacketBatch *batch) {
...
-      bytes_cnt_ += batch->pkts()[i]->total_len();
-      total_latency_ += diff;
+        bytes_cnt_ += batch->pkts()[i]->total_len();
+        total_latency_ += diff;
 
-      record_latency(&hist_, diff);
+        record_latency(&hist_, diff);
+      }
     }
+    RunNextModule(batch);
```
Comment (melvinw) [inline core/modules/measure.cc]: Why call `RunNextModule` twice? Is this behavior defined?

Comment (apanda) [inline core/modules/measure.cc]: Oops, it was a merge conflict, sorry I did not build before submitting (serves me right). Let me fix it and update it shortly.

**Comment by apanda** [comment]
Fixed those comments, sorry.

**Comment by melvinw** [comment]
Thanks! Looks good to me otherwise. Will approve after CI passes

**Comment by melvinw** [comment]
Thanks! 
cat

**Pull request merged**